src/legacy/ui/ui_render/ui_render_mixin.js

```diff
@@ -192,121 +176,41 @@ export function uiRenderMixin(kbnServer, server, config) {
     },
   });
 
-  async function getUiSettings({ request, includeUserProvidedConfig }) {
-    const uiSettings = request.getUiSettingsService();
-    return props({
-      defaults: uiSettings.getRegistered(),
-      user: includeUserProvidedConfig && uiSettings.getUserProvided(),
-    });
-  }
-
-  function getLegacyKibanaPayload({ app, basePath, uiSettings }) {
-    return {
-      app,
-      bundleId: `app:${app.getId()}`,
-      nav: server.getUiNavLinks(),
-      version: kbnServer.version,
-      branch: config.get('pkg.branch'),
-      buildNum: config.get('pkg.buildNum'),
-      buildSha: config.get('pkg.buildSha'),
-      serverName: config.get('server.name'),
-      devMode: config.get('env.dev'),
-      basePath,
-      uiSettings,
-    };
-  }
-
-  async function renderApp({
-    app,
+  async function renderApp(
     h,
-    includeUserProvidedConfig = true,
-    injectedVarsOverrides = {},
-  }) {
-    const request = h.request;
-    const basePath = request.getBasePath();
-    const uiSettings = await getUiSettings({ request, includeUserProvidedConfig });
-    app = app || { getId: () => 'core' };
-
-    const legacyMetadata = getLegacyKibanaPayload({
-      app,
-      basePath,
-      uiSettings,
+    app = { getId: () => 'core' },
+    includeUserSettings = true,
+    overrides = {}
+  ) {
+    const { http } = kbnServer.newPlatform.setup.core;
+    const {
+      rendering,
+      legacy,
+      savedObjectsClientProvider: savedObjects,
+      uiSettings: { asScopedToClient },
+    } = kbnServer.newPlatform.__internals;
+    const uiSettings = asScopedToClient(savedObjects.getClient(h.request));
+    const vars = await legacy.getVars(app.getId(), h.request, {
+      apmConfig: getApmConfig(app),
```

Comment:
as I understand it expects appMetadata, not the app. https://github.com/elastic/kibana/blob/d835eaca4b4fdef4eb66c3bc3445a1e5cbcebe6c/src/legacy/ui/apm/index.js#L33 therefore it will throw on the NP app https://github.com/elastic/kibana/blob/d835eaca4b4fdef4eb66c3bc3445a1e5cbcebe6c/src/legacy/ui/apm/index.js#L41

Reply:
The previous version used this exact variable.

https://github.com/elastic/kibana/blob/3ed5264cc307964ced5f90402264232724956602/src/legacy/ui/ui_render/ui_render_mixin.js#L287

cc: @vigneshshanmugam